src/averageAges.js

```diff
@@ -1,61 +1,53 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function sumAges(people) {
+  return people.reduce((sum, person) => sum + (person.died - person.born), 0);
+}
+
+function isMother(people) {
+  return people.filter(women => women.sex === 'f'
+    && people.some(person => person.mother === women.name));
+}
+
+function isSon(people, mothers) {
+  return people.filter(child => child.sex === 'm'
+  && mothers.find(mother => mother.name === child.mother));
+}
+
+function isChild(people, mothers) {
+  return people.filter(child =>
+    mothers.find(mother => mother.name === child.mother));
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredByCentury = century
+    ? people.filter(person => Math.ceil(person.died / 100) === century)
+    : people;
+
+  const filteredBySex = filteredByCentury.filter(person => person.sex === 'm');
```

Comment:
use destructure assignment where it's possible. check throughout your code

```suggestion
  const filteredBySex = filteredByCentury.filter(({ sex })=> sex === 'm');
```